guidance/models/transformers/_transformers.py

```diff
@@ -49,81 +53,207 @@ def __init__(
         **kwargs,
     ):
         if transformers_tokenizer is None:
-            transformers_tokenizer = self._tokenizer(model, **kwargs)
+            if isinstance(model, str):
+                transformers_tokenizer = self._tokenizer(model, **kwargs)
+            else:
+                raise ValueError(
+                    "A model object was passed in, but no tokenizer was provided. Please provide a tokenizer."
+                )
         else:
             is_ptt = isinstance(transformers_tokenizer, transformers_package.PreTrainedTokenizer)
             is_ptt_fast = isinstance(
                 transformers_tokenizer, transformers_package.PreTrainedTokenizerFast
             )
             assert is_ptt or is_ptt_fast
-
         self._orig_tokenizer = transformers_tokenizer
-        special_tokens_map = {
-            id: token for token, id in transformers_tokenizer.get_added_vocab().items()
-        }
+        byte_tokens = self._byte_tokens(transformers_tokenizer)
 
-        # build out the set of byte_string tokens
-        byte_tokens = [b""] * len(transformers_tokenizer)
-        if hasattr(transformers_tokenizer, "byte_decoder"):
-            byte_decoder = transformers_tokenizer.byte_decoder
+        # Chat Template logic
+        if chat_template is None and hasattr(self._orig_tokenizer, "chat_template"):
+            chat_template = self._orig_tokenizer.chat_template
 
-            for i in range(len(transformers_tokenizer)):
-                byte_coded = bytes(
-                    [byte_decoder[c] for c in transformers_tokenizer.convert_ids_to_tokens(i)]
+        # the superclass does most of the work once we have the tokens
+        super().__init__(
+            byte_tokens,
+            chat_template,
+            None if ignore_bos_token else transformers_tokenizer.bos_token_id,
+            transformers_tokenizer.eos_token_id,
+        )
+
+    def _tokenizer(self, model: str, **kwargs) -> Union[
+        "transformers_package.PreTrainedTokenizer",
+        "transformers_package.PreTrainedTokenizerFast",
+    ]:
+        # make sure transformers is installed
+        if not has_transformers:
+            raise ImportError("Please install transformers with `pip install transformers`")
+
+        try:
+            tokenizer = transformers_package.AutoTokenizer.from_pretrained(
+                model, use_fast=False, **kwargs
+            )
+        except ImportError:
```

Comment:
Is there any point to this `except` if all it does is re-raise?

Reply:
Ahh sorry, it's because of the generic 'exception' catch which is next. Perhaps add a comment that you're filtering out the `ImportErrors` first?

Reply:
Added a comment!